routes/router.js

```diff
@@ -1,58 +1,232 @@
+var router = express.Router();
+var User = require('../model/User');
+var bodyParser = require('body-parser');
+var alert = require('alert-node');
+const moment = require('moment');
+const pathName = require('path');
+const json2csv = require('json2csv').parse;
+const fields = ['imagePath', 'imagesBinary', 'name', 'folio',
+    'description', 'classification', 'mei', 'review', 'dob',
+    'project', 'neumeSection', 'neumeSectionName'
+];
+global.userArray = [];
+global.userArray = [];
+var dialog = require('dialog');
+var mongoose = require('mongoose');
+var async = require('async');
+var crypto = require('crypto');
+var flash = require('express-flash');
+var storedImages = require('../model/storedImages');
+var SENDGRID_API_KEY = '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';
 var logger = require('../logger');
 
-router.use(flash())
+router.use(flash());
+
 router.use(bodyParser.json({
     limit: '50mb'
-}))
+}));
+
 router.use(bodyParser.urlencoded({
     limit: '50mb',
     extended: true
-}))
-// GET route for reading data
+}));
+
+// necessary stuff for file IO
+var multer = require('multer')
+var uploadCSV = multer({
+    dest: 'exports/'
+})
+
+var fs = require('fs');
+var dir = './exports';
+
+if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir);
+}
+var multer = require('multer')
+var uploadCSV = multer({
+    storage: multer.diskStorage({
+        destination: function(req, file, callback) {
+            callback(null, './exports');
+        },
+        filename: function(req, file, callback) {
+            callback(null, file.originalname);
+        }
+    })
+}).single('fileImage');
+
+var userFinal = [];
+
+function renderError(res, err) {
+    logger.error(res);
+    logger.error(err);
+    return res.format({
+        html: function() {
+            res.render('errorLog', {
+                "error": err,
+            });
+        },
+        json: function() {
+            res.json(err);
+        }
+    });
+}
+
+// load the index page, and flag that we're logged out
 router.get('/', function(req, res, next) {
+    req.session.userId = -1;
     return res.render('index')
-})
-var userFinal = []
+});
 
-/*Forget Password Page*/
+//POST route for updating data
+router.post('/', function(req, res, next) {
+    var editor = false;
+    req.session.userId = -1; // set logged out
+
+    if (req.body.email &&
+        req.body.username &&
+        req.body.password &&
+        req.body.passwordConf) {
+
+        // confirm that user typed same password twice
+        if (req.body.password !== req.body.passwordConf) {
+            var err = new Error('Passwords do not match. Try again');
+            return renderError(res, err);
+        }
 
+        var userData = {
+            email: req.body.email,
+            username: req.body.username,
+            password: req.body.password,
+            role: req.body.role,
+            bio: req.body.bio,
+            active: false
+        }
```

Comment:
Is any of this data checked? For example, that the email is an email, or certain characters for the username & password, or role?

Reply:
I've opened an issue for the problem with the login system, I think it's worth making another branch and dealing with it there